docs/aws/audit/cloudfrontmonitoring/rules/cloudfront_distributions_https.mdx

```diff
@@ -23,6 +23,262 @@ HITRUST, SOC2, NISTCSF, PCIDSS
+
+   Installation:
+   ```
+   pip install awscli
+   ```
+   Configuration:
+   ```
+   aws configure
+   ```
+   You will be prompted to provide your AWS Access Key ID, Secret Access Key, Default region name, and Default output format.
+
+2. List all CloudFront distributions: Use the following AWS CLI command to list all the CloudFront distributions in your AWS account:
+
+   ```
+   aws cloudfront list-distributions --query 'DistributionList.Items[*].Id' --output text
+   ```
+   This command will return a list of IDs for all your CloudFront distributions.
+
+3. Check HTTPS configuration for each distribution: For each distribution ID returned by the previous command, run the following command to check its HTTPS configuration:
+
+   ```
+   aws cloudfront get-distribution-config --id <distribution-id>
+   ```
+   Replace `<distribution-id>` with the actual ID of your CloudFront distribution. This command will return the configuration details of the specified distribution.
+
+4. Check the ViewerProtocolPolicy: In the output of the previous command, look for the `ViewerProtocolPolicy` field under `DefaultCacheBehavior`. If the value of this field is `allow-all`, it means that both HTTP and HTTPS are enabled on the distribution. If the value is `https-only`, it means that only HTTPS is enabled.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install and configure AWS SDK for Python (Boto3). You will need to set up your AWS credentials and AWS region to use Boto3. You can install Boto3 using pip:
+
+```python
+pip install boto3
+```
+
+2. Import the necessary modules and create a session using your credentials:
+
+```python
+import boto3
+
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
+    region_name='YOUR_REGION'
+)
+```
+
+3. Create a CloudFront client and retrieve all distributions:
+
+```python
+cf = session.client('cloudfront')
+distributions = cf.list_distributions()
+```
+
+4. Iterate over each distribution and check if HTTPS is enabled:
+
+```python
+for distribution in distributions['DistributionList']['Items']:
+    if 'ViewerCertificate' in distribution and 'MinimumProtocolVersion' in distribution['ViewerCertificate']:
+        if distribution['ViewerCertificate']['MinimumProtocolVersion'] != 'TLSv1.2_2018':
+            print(f"CloudFront distribution {distribution['Id']} does not have HTTPS enabled.")
+```
+
+This script will print out the IDs of all CloudFront distributions that do not have HTTPS enabled. The 'MinimumProtocolVersion' attribute is used to determine if HTTPS is enabled, as it specifies the minimum SSL/TLS protocol that can be used to access the content. If it's not set to 'TLSv1.2_2018' or higher, then HTTPS is not enabled.
+</Accordion>
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>